src/admin/admin.service.spec.ts

```diff
@@ -1,20 +1,84 @@
 import type { TestingModule } from '@nestjs/testing';
 import { Test } from '@nestjs/testing';
+import type { MockProxy } from 'jest-mock-extended';
+import { mockDeep } from 'jest-mock-extended';
 
+import { PrismaService } from '../prisma/prisma.service';
 import { AdminService } from './admin.service';
+import {
+  adminExpectedResponse,
+  adminStub,
+  loginInputInValidPasswordDtoStub,
+  loginInputValidPasswordDtoStub,
+} from './stubs';
 
 describe('AdminService', () => {
   let service: AdminService;
+  let prisma: MockProxy<PrismaService>;
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      providers: [AdminService],
+      providers: [
+        AdminService,
+        {
+          provide: PrismaService,
+          useFactory: () => mockDeep<PrismaService>(),
+        },
+      ],
     }).compile();
 
     service = module.get<AdminService>(AdminService);
+    prisma = module.get(PrismaService);
   });
 
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
+
+  describe('localLogin', () => {
+    it('should be defined', () => {
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(service.localLogin).toBeDefined();
+    });
+
+    it('should be called with', async () => {
+      const test = jest
+        .spyOn(service, 'localLogin')
+        .mockResolvedValue({} as any);
+
+      await service.localLogin(loginInputInValidPasswordDtoStub);
+
+      expect(test).toHaveBeenCalledWith(loginInputInValidPasswordDtoStub);
+    });
+
+    it('should throw an error: no admin found with this username', async () => {
```

Comment:
it doesn't throw error in case that no admin found, it'll return null

Reply:
fixed